src/Nuget.PackageIndex.VisualStudio/DnxProjectFilter.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using EnvDTE;
+using EnvDTE80;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Nuget.PackageIndex.Client.Analyzers;
+
+namespace Nuget.PackageIndex.VisualStudio
+{
+    /// <summary>
+    /// This filter is needed only until we enable Missing Package suggestions for all C# projects.
+    /// For now we limit this feature only to ProjectK scenarios.
+    /// </summary>
+    internal class DnxProjectFilter : IProjectFilter, IDisposable
+    {
+        private ConcurrentDictionary<string, string> FilesCache = new ConcurrentDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private bool _disposed;
+        private SolutionEvents _solutionEvents;
+        private object _cacheLock = new object();
+        private bool _solutionIsClosing;
+
+        public DnxProjectFilter()
+        {
+            var container = ServiceProvider.GlobalProvider.GetService<IComponentModel, SComponentModel>();
```

Comment:
An alternative is to move initialization off the constructor. In some cases its the right move, not convinced if it is necessary here. Just a thought.

-----Original Message-----
From: "Anton Piskunov" <notifications@github.com>
Sent: ‎5/‎6/‎2015 3:28 PM
To: "NuGet/NuGet.PackageIndex" <NuGet.PackageIndex@noreply.github.com>
Cc: "Yishai Galatzer" <yishai_galatzer@yahoo.com>
Subject: Re: [NuGet.PackageIndex] Antonpis/tests (#6)

In src/Nuget.PackageIndex.VisualStudio/DnxProjectFilter.cs:
> +{
> +    /// <summary>
> +    /// This filter is needed only until we enable Missing Package suggestions for all C# projects.
> +    /// For now we limit this feature only to ProjectK scenarios.
> +    /// </summary>
> +    internal class DnxProjectFilter : IProjectFilter, IDisposable
> +    {
> +        private ConcurrentDictionary<string, string> FilesCache = new ConcurrentDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
> +        private bool _disposed;
> +        private SolutionEvents _solutionEvents;
> +        private object _cacheLock = new object();
> +        private bool _solutionIsClosing;
> +
> +        public DnxProjectFilter()
> +        {
> +            var container = ServiceProvider.GlobalProvider.GetService<IComponentModel, SComponentModel>();
It must be on UI thread (by Roslyn), but you are right, we have a helper to throw and I will add it there
—
Reply to this email directly or view it on GitHub.